cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,376 @@
+# Radius AWS Workload Identity Support
+
+* **Status**: In Review
+* **Author**: Nithya Subramanian (@nithyatsu)
+
+## Overview
+
+A software workload such as a container-based application, service or script needs an identity to authenticate, access, and communicate with services that are distributed across different platforms and/or cloud providers. Radius uses static credentials (Access Key) to interact with AWS IAM (Identity and Access Management) so that we can deploy and access AWS resources today:
+ 
+```
+rad credential register aws --access-key-id <access-key-id> --secret-access-key <secret-access-key>
+```
+ 
+These IAM credentials, including the access key should be rotated regularly to reduce the chance of unauthorized access. A more secure option than using these credentials is to use IRSA.
+ 
+IRSA (IAM Roles for Service Accounts) is used for authenticating applications running within Kubernetes pods. When we use IRSA, we associate an IAM role directly with a Kubernetes service-account, allowing the pods associated with the service-account to assume that role. IRSA does not require us to configure access key and  secret. It relies identity tokens (OIDC tokens) to authenticate with AWS services. These tokens have a default expiration (configurable). However, they automatically refresh when needed, transparent to the user.
+
+The goal of the scenario is to enable infrastructure operators to configure IRSA support for the AWS provider in Radius to deploy and manage AWS resources.
+
+## Terms and definitions
+| Term | Definition |
+|---|---|
+| OIDC | OIDC stands for OpenID Connect. It allows authentication of end-user(or workload)|
+|Role ARN| ARN associated with an AWS IAM role|
+
+
+## Objectives
+
+> Issue Reference: https://github.com/radius-project/radius/issues/7618 
+
+### Goals
+
+* Radius users can configure AWS provider and enable Radius to use IRSA for authentication and deployment of application.
+* IRSA can be configured via interactive experience.
+* IRSA can be configured with non-interactive commands.
+* Radius users can deploy and manage AWS resources without having to manage AWS access key id and secret
+
+### Non-goals
+
+* Azure Managed Identity support
+* Azure Workload Identity support
+
+The above two are addressed in [Azure Workload Identitu Support](https://github.com/radius-project/design-notes/blob/main/cli/2024-04-azure-workload-identity.md)
```

Comment:
nit

```suggestion
The above two are addressed in [Azure Workload Identity Support](https://github.com/radius-project/design-notes/blob/main/cli/2024-04-azure-workload-identity.md)
```

Reply:
+1 . The link seems broken also.

Reply:
fixed the spelling. The design note is still in PR. I think this would be the link once it is merged cc @willdavsmith . I will check it again at that point.